File: API/Models/Aircraft.cs
using System;

namespace API.Models;

public class Aircraft
{
    public int Id { get; set; }
    public required string Model { get; set; }
    public int TotalSeats { get; set; }
    public int SeatRows { get; set; }
    public int SeatsPerRow { get; set; }

    public List<Seat> Seats { get; set; } = [];
    public List<Flight> Flights { get; set; } = [];
}

File: client/src/models/aircraft.ts
export interface AircraftOption {
    id: number;
    model: string;
    registrationNumber: string;
    totalSeats: number;
    seatRows: number;
    seatsPerRow: number;
}

export interface CreateAircraftDto {
    model: string;
    registrationNumber: string;
    totalSeats: number;
    seatRows: number;
    seatsPerRow: number;
}

export interface UpdateAircraftDto {
    model: string;
    registrationNumber: string;
    totalSeats: number;
    seatRows: number;
    seatsPerRow: number;
}

export interface AircraftOperationResult {
    succeeded: boolean;
    aircraft: AircraftOption | null;
    errors: string[];
}
